unyt/_array_functions.py

```diff
@@ -469,10 +469,9 @@ def _array_comp_helper(a, b):
     au = getattr(a, "units", NULL_UNIT)
     bu = getattr(b, "units", NULL_UNIT)
     if bu != au and au != NULL_UNIT and bu != NULL_UNIT:
-        if (bu / au).is_dimensionless:
-            b = np.array(b) * (1 * bu).to(au)
-        else:
-            raise UnitConversionError(au, au.dimensions, bu, bu.dimensions)
+        if not bu.same_dimensions_as(au):
+            UnitConversionError(au, au.dimensions, bu, bu.dimensions)
```

Comment:
this needs a `raise` statement

Reply:
and i guess a test for the exception case :stuck_out_tongue:

Reply:
How did I miss this ???

Reply:
interestingly, this test already exists *and* already passes because the error is raised from the following line when conversion fails.
The test has a typo in its name though, so I'll fix it, and remove this redundant (and broken !) branch.